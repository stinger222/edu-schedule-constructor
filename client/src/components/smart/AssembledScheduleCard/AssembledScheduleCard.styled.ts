import styled from "styled-components"

export const StyledAssembledSchedule = styled.div`
	width: 37em;
	height: fit-content;
	padding: 0 0.3em 1em 0.3em;
  position: relative;

	border-radius: 1.3em;
	background: ${({theme}) => theme.assembledScheduleCard.background};
	box-shadow: ${({theme}) => theme.assembledScheduleCard.boxShadow};
	
  & > header {
    position: relative;
    padding-block: 1.2em;
  }

  & > header > input.switch {
    position: absolute;
    right: 0.5em;
    top: 50%;
    transform: translateY(-55%);
  }

	& > header .title {
		line-height: 1em;
		font-size: 1.9em;
		font-weight: 400;
		letter-spacing: 0.05em;
		text-align: center;
    margin-inline: auto;
    width: 77%;
	}

	& .days {
		display: flex;
		flex-wrap: wrap;
		justify-content: center;
		gap: 0.6em;
		row-gap: 0.7em;
	}

	& .days > * {
		flex-basis: calc(33% - 0.4em);
	}

  @media(min-width: 700px) {
    & > header > input.switch {
      font-size: 0.8em;
    }
  }
`