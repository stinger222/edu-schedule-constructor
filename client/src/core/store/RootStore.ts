import { configure } from "mobx"
import { IAssembledSchedulesStore, IAuthStore, IClassesStore, IClassSchedulesStore, IUIStore } from "./../types/store"
import AssembledSchedulesStore from "./AssembledSchedulesStore"
import ClassSchedulesStore from "./ClassSchedulesStore"
import ClassesStore from "./ClassesStore"
import UIStore from "./UIStore"
import AuthStore from "./AuthStore"

class RootStore {
	uiStore: IUIStore
	classesStore: IClassesStore
	classSchedulesStore: IClassSchedulesStore
	assembledSchedulesStore: IAssembledSchedulesStore
  authStore: IAuthStore

	constructor() {
		this.uiStore = new UIStore()
		this.classesStore = new ClassesStore()
		this.classSchedulesStore = new ClassSchedulesStore()
		this.assembledSchedulesStore = new AssembledSchedulesStore()
    this.authStore = new AuthStore()
	}
}

configure({
  enforceActions: "never"
})

export default RootStore