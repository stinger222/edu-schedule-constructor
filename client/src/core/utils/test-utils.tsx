import React, { ReactElement } from "react"
import { HashRouter } from "react-router-dom"
import { ThemeProvider } from "styled-components"
import { render, RenderOptions } from "@testing-library/react"
import { DarkTheme } from "../style/themes/Dark"
import MockRootStore from "../store/__mocks__/MockRootStore"
import "../configs/i18next"

const AllTheProviders = ({ children }: { children: ReactElement }) => {
	const rootStore = new MockRootStore()
	const TestStoreContext = React.createContext<typeof rootStore>(rootStore)

	return (
		<ThemeProvider theme={DarkTheme}>
			<TestStoreContext.Provider value={rootStore}>
				<HashRouter>
					{children}
				</HashRouter>
			</TestStoreContext.Provider>
		</ThemeProvider>
	)
}

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, "queries">) => {
	return render(ui, { wrapper: AllTheProviders, ...options})
}

export * from "@testing-library/react"
export { customRender as render }
