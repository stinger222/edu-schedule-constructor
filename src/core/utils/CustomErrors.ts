export class EmptyDay extends Error {
  readonly name = "EmptyDay"

  constructor() {
    super()
  }
}

export class Warning extends Error {
  readonly name = "Warning"

  constructor(message: string) {
    super(message)
  }
}
