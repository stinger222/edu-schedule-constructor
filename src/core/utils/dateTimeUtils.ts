import { Cases, weekDates } from "../types/types"
import { formatNumber } from "./stringUtils"


interface IData {
  ru: {
    short: readonly string[],
    full: {
      nominative: readonly string[],
      accusative: readonly string[]
    }
  },
  en: {
    short: readonly string[],
    full: {
      nominative: readonly string[],
      accusative: readonly string[]
    }
  }
}
export class WeekDays {

  private static readonly data: IData = {
    ru: {
      short: ["Пн", "Вт", "Ср", "Чт", "Пт", "Сб", "Вс"],
      full: {
        nominative: [
          "Понедельник", "Вторник", "Среда", "Четверг", "Пятница", "Суббота", "Воскресенье"
        ],
        accusative: [
          "Понедельник", "Вторник", "Среду", "Четверг", "Пятницу", "Субботу", "Воскресенье"
        ]
      }
    },
    en: {
      short: ["M", "T", "W", "T", "F", "S", "S"],
      full: {
        nominative: [
          "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"
        ],
        accusative: [
          "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"
        ]
      }
    }
  }

	static getShort(lang: "ru" | "en"): ReadonlyArray<string> {
		return this.data[lang].short as ReadonlyArray<string>
	}

	static getFull(lang: "ru" | "en", caseType: Cases = Cases.Nominative, toLowerCase: boolean = false): ReadonlyArray<string> {
		const result: ReadonlyArray<string> = this.data[lang].full[caseType]
		return toLowerCase ? result.map(i => i.toLowerCase()) : result
	}
}

export const getCurrentWeekDates = (): weekDates => {
	const date = new Date()
	date.setDate(date.getDate() - (date.getDay() === 0 ? 7 : date.getDay())) // now it's current week sunday date:

	const result = Array.from({ length: 7 }, () => {
		date.setDate(date.getDate() + 1) // and now it's monday
		return formatNumber(date.getDate())
	})

	return result as weekDates
}
