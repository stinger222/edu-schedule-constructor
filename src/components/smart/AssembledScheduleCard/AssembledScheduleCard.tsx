import { useContext } from "react"
import { StoreContext } from "../../.."
import { IAssembledDay } from "../../../core/types/types"
import { StyledAssembledSchedule } from "./AssembledScheduleCard.styled"

import Day from "./Day/Day"
import Switch from "../../ui/Switch/Switch"
import { observer } from "mobx-react"

interface IProps {
	name: string,
	days: IAssembledDay[]
  uid: string
}

const AssembledScheduleCard: React.FC<IProps> = ({ name, days, uid }) => {

  const { assembledSchedulesStore } = useContext(StoreContext)
  const thisIsActive = assembledSchedulesStore.activeScheduleUid === uid

  const handleActivation = () => {
    assembledSchedulesStore.activateSchedule(uid)
  }

  return (
		<StyledAssembledSchedule className="assembled-schedule">
			<header>
        <div className="title">{ name }</div>
        <Switch type="checkbox" onChange={handleActivation} checked={thisIsActive} />
      </header>
      <div className="days">
				{
					new Array(5).fill(0).map((_, index) => {
						const day = days[index]
						return (
							<Day
								dayIndex={index}
								day={day}
								key={index}
							/>
						)
					})
				}
			</div>
		</StyledAssembledSchedule>
	)
}

export default observer(AssembledScheduleCard)
