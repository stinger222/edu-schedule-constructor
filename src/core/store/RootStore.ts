import { configure } from "mobx"
import { IComposedSchedulesStore, ILessonsStore, IClassSchedulesStore, IUIStore } from "./../types/store"
import ComposedSchedulesStore from "./ComposedSchedulesStore"
import ClassSchedulesStore from "./ClassSchedulesStore"
import LessonsStore from "./LessonsStore"
import UIStore from "./UIStore"
class RootStore {
	uiStore: IUIStore
	lessonsStore: ILessonsStore
	classSchedulesStore: IClassSchedulesStore
	composedSchedulesStore: IComposedSchedulesStore

	constructor() {
		this.uiStore = new UIStore()
		this.lessonsStore = new LessonsStore()
		this.classSchedulesStore = new ClassSchedulesStore()
		this.composedSchedulesStore = new ComposedSchedulesStore()
	}
}

configure({
    enforceActions: "never"
})

export default RootStore