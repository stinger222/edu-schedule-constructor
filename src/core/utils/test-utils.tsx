import React, { ReactElement } from "react"
import { HashRouter } from "react-router-dom"
import { render, RenderOptions } from "@testing-library/react"

import RootStore from "../store/RootStore"


import { ThemeProvider } from "styled-components"
import { DarkTheme } from "../themes/Dark"

const AllTheProviders: React.FC<{children: ReactElement}> = ({children}) => {

	const rootStore = new RootStore()
	const StoreContext = React.createContext<typeof rootStore>(rootStore)

	return (
		<ThemeProvider theme={DarkTheme}>
			<StoreContext.Provider value={rootStore}>
				<HashRouter>
					{children}
				</HashRouter>
			</StoreContext.Provider>
		</ThemeProvider>
	)
}

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, "queries">) => {
	return render(ui, { wrapper: AllTheProviders, ...options})
}

export * from "@testing-library/react"
export {customRender as render}