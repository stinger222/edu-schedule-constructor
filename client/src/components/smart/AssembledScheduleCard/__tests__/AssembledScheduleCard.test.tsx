import { renderHook } from "@testing-library/react"
import { useContext } from "react"
import { StoreContext } from "../../../.."
import { render } from "../../../../core/utils/test-utils"
import RootStore from "../../../../core/store/RootStore"
import AssembledScheduleCard from "../AssembledScheduleCard"

let rootStore: RootStore
describe("Testing AssembledScheduleCard component", () => {
  beforeAll(() => {
    rootStore = renderHook(() => useContext(StoreContext)).result.current
    
    rootStore.classSchedulesStore.classSchedules = []
    rootStore.classSchedulesStore.classSchedules.push({
      name: "Test Class Schedule (4 classes described)",
      classes: [
        {start: "08:30", end: "10:00"},
        {start: "10:30", end: "12:00"},
        {start: "12:30", end: "14:00"},
        {start: "14:30", end: "16:00"}
      ],
      uid: "test-classes-schedule"
    })
  })

  it("Properly renders schedule card", () => {
    const screen = render(
      <AssembledScheduleCard
        name="Test Schedule #1"
        days={[
          {classScheduleId: "test-classes-schedule", classIds: ["1", "2", "3", "4"]},
          {classScheduleId: "test-classes-schedule", classIds: ["hidden", "2", "3", "4"]},
          {classScheduleId: "test-classes-schedule", classIds: ["hidden", "hidden", "3", "4"]},
          {classScheduleId: "test-classes-schedule", classIds: ["hidden"]},
          {classScheduleId: "test-classes-schedule", classIds: ["hidden", "hidden", "hidden", "hidden"]}
        ]}
        uid="test-assembled-schedule-id-1"
      />
    )
    expect(screen).toMatchSnapshot()
  })
})