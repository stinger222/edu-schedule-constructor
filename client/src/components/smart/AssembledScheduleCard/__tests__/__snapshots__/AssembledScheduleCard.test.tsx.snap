// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing AssembledScheduleCard component > Properly renders schedule card 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c2 {
  font-size: 1.4em;
  width: 8.4em;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  border-radius: 0.8em;
  padding-bottom: 0.5em;
  background: #37393F;
  box-shadow: 0.1em 0.2em 0.2em 0 #00000080;
}

.c2 header {
  font-size: 1.3em;
  line-height: 1em;
  font-weight: 400;
  text-align: center;
  padding-block: 0.5em;
}

.c2 .card-body {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 1fr 1fr 1fr;
  padding-inline: 0.3em;
}

.c2 .card-body span {
  text-align: center;
  line-height: 1.5em;
}

.c2 .card-body span:nth-child(2n - 1) {
  grid-column: 1 / 2;
  font-weight: 100;
}

.c2 .card-body span:nth-child(2n) {
  grid-column: 2 / 3;
  font-weight: 400;
}

.c1 {
  --transition-duration: 0.3s;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  font-size: 1em;
  width: 4em;
  height: 2em;
  box-shadow: inset 0 0 0.4em #00000060;
  border-radius: 10em;
  background: #50535a;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

.c1:after {
  content: "";
  position: absolute;
  top: 0em;
  left: 0;
  width: 2em;
  height: 2em;
  border-radius: 10em;
  box-shadow: 0 0 0.4em 0em #00000060;
  background: white;
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
  -webkit-transition: all var(--transition-duration);
  transition: all var(--transition-duration);
}

.c1:checked {
  background: #0068e2;
  box-shadow: none;
}

.c1:checked::after {
  left: 100%;
  -webkit-transform: translateX(-100%) scale(1.05);
  -ms-transform: translateX(-100%) scale(1.05);
  transform: translateX(-100%) scale(1.05);
}

.c0 {
  width: 37em;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 0 0.3em 1em 0.3em;
  position: relative;
  border-radius: 1.3em;
  background: #2A2C32;
  box-shadow: 0.4em 0.4em 0.4em 0 #00000080;
}

.c0 > header {
  position: relative;
  padding-block: 1.2em;
}

.c0 > header > input.switch {
  position: absolute;
  right: 0.5em;
  top: 50%;
  -webkit-transform: translateY(-55%);
  -ms-transform: translateY(-55%);
  transform: translateY(-55%);
}

.c0 > header .title {
  line-height: 1em;
  font-size: 1.9em;
  font-weight: 400;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  text-align: center;
  margin-inline: auto;
  width: 77%;
}

.c0 .days {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.6em;
  row-gap: 0.7em;
}

.c0 .days > * {
  -webkit-flex-basis: calc(33% - 0.4em);
  -ms-flex-preferred-size: calc(33% - 0.4em);
  flex-basis: calc(33% - 0.4em);
}

@media (min-width:700px) {
  .c0 > header > input.switch {
    font-size: 0.8em;
  }
}

<body>
    <div />
    <div>
      <div
        class="c0 assembled-schedule"
      >
        <header>
          <div
            class="title max-lines-2"
          >
            Test Schedule #1
          </div>
          <input
            class="c1 switch"
            type="checkbox"
          />
        </header>
        <div
          class="days"
        >
          <div
            class="c2"
          >
            <header>
              M
            </header>
            <div
              class="card-body"
            >
              <span>
                 
                Classes
                 
              </span>
              <span>
                4
              </span>
              <span>
                 
                Start
                 
              </span>
              <span>
                08:30
              </span>
              <span>
                 
                End
                 
              </span>
              <span>
                16:00
              </span>
            </div>
          </div>
          <div
            class="c2"
          >
            <header>
              T
            </header>
            <div
              class="card-body"
            >
              <span>
                 
                Classes
                 
              </span>
              <span>
                3
              </span>
              <span>
                 
                Start
                 
              </span>
              <span>
                10:30
              </span>
              <span>
                 
                End
                 
              </span>
              <span>
                16:00
              </span>
            </div>
          </div>
          <div
            class="c2"
          >
            <header>
              W
            </header>
            <div
              class="card-body"
            >
              <span>
                 
                Classes
                 
              </span>
              <span>
                2
              </span>
              <span>
                 
                Start
                 
              </span>
              <span>
                12:30
              </span>
              <span>
                 
                End
                 
              </span>
              <span>
                16:00
              </span>
            </div>
          </div>
          <div
            class="c2"
          >
            <header>
              T
            </header>
            <div
              class="card-body"
            >
              <span>
                 
                Classes
                 
              </span>
              <span>
                0
              </span>
              <span>
                 
                Start
                 
              </span>
              <span>
                -:-
              </span>
              <span>
                 
                End
                 
              </span>
              <span>
                -:-
              </span>
            </div>
          </div>
          <div
            class="c2"
          >
            <header>
              F
            </header>
            <div
              class="card-body"
            >
              <span>
                 
                Classes
                 
              </span>
              <span>
                0
              </span>
              <span>
                 
                Start
                 
              </span>
              <span>
                -:-
              </span>
              <span>
                 
                End
                 
              </span>
              <span>
                -:-
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c2 {
  font-size: 1.4em;
  width: 8.4em;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  border-radius: 0.8em;
  padding-bottom: 0.5em;
  background: #37393F;
  box-shadow: 0.1em 0.2em 0.2em 0 #00000080;
}

.c2 header {
  font-size: 1.3em;
  line-height: 1em;
  font-weight: 400;
  text-align: center;
  padding-block: 0.5em;
}

.c2 .card-body {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 1fr 1fr 1fr;
  padding-inline: 0.3em;
}

.c2 .card-body span {
  text-align: center;
  line-height: 1.5em;
}

.c2 .card-body span:nth-child(2n - 1) {
  grid-column: 1 / 2;
  font-weight: 100;
}

.c2 .card-body span:nth-child(2n) {
  grid-column: 2 / 3;
  font-weight: 400;
}

.c1 {
  --transition-duration: 0.3s;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  font-size: 1em;
  width: 4em;
  height: 2em;
  box-shadow: inset 0 0 0.4em #00000060;
  border-radius: 10em;
  background: #50535a;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

.c1:after {
  content: "";
  position: absolute;
  top: 0em;
  left: 0;
  width: 2em;
  height: 2em;
  border-radius: 10em;
  box-shadow: 0 0 0.4em 0em #00000060;
  background: white;
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
  -webkit-transition: all var(--transition-duration);
  transition: all var(--transition-duration);
}

.c1:checked {
  background: #0068e2;
  box-shadow: none;
}

.c1:checked::after {
  left: 100%;
  -webkit-transform: translateX(-100%) scale(1.05);
  -ms-transform: translateX(-100%) scale(1.05);
  transform: translateX(-100%) scale(1.05);
}

.c0 {
  width: 37em;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 0 0.3em 1em 0.3em;
  position: relative;
  border-radius: 1.3em;
  background: #2A2C32;
  box-shadow: 0.4em 0.4em 0.4em 0 #00000080;
}

.c0 > header {
  position: relative;
  padding-block: 1.2em;
}

.c0 > header > input.switch {
  position: absolute;
  right: 0.5em;
  top: 50%;
  -webkit-transform: translateY(-55%);
  -ms-transform: translateY(-55%);
  transform: translateY(-55%);
}

.c0 > header .title {
  line-height: 1em;
  font-size: 1.9em;
  font-weight: 400;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  text-align: center;
  margin-inline: auto;
  width: 77%;
}

.c0 .days {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.6em;
  row-gap: 0.7em;
}

.c0 .days > * {
  -webkit-flex-basis: calc(33% - 0.4em);
  -ms-flex-preferred-size: calc(33% - 0.4em);
  flex-basis: calc(33% - 0.4em);
}

@media (min-width:700px) {
  .c0 > header > input.switch {
    font-size: 0.8em;
  }
}

<div>
    <div
      class="c0 assembled-schedule"
    >
      <header>
        <div
          class="title max-lines-2"
        >
          Test Schedule #1
        </div>
        <input
          class="c1 switch"
          type="checkbox"
        />
      </header>
      <div
        class="days"
      >
        <div
          class="c2"
        >
          <header>
            M
          </header>
          <div
            class="card-body"
          >
            <span>
               
              Classes
               
            </span>
            <span>
              4
            </span>
            <span>
               
              Start
               
            </span>
            <span>
              08:30
            </span>
            <span>
               
              End
               
            </span>
            <span>
              16:00
            </span>
          </div>
        </div>
        <div
          class="c2"
        >
          <header>
            T
          </header>
          <div
            class="card-body"
          >
            <span>
               
              Classes
               
            </span>
            <span>
              3
            </span>
            <span>
               
              Start
               
            </span>
            <span>
              10:30
            </span>
            <span>
               
              End
               
            </span>
            <span>
              16:00
            </span>
          </div>
        </div>
        <div
          class="c2"
        >
          <header>
            W
          </header>
          <div
            class="card-body"
          >
            <span>
               
              Classes
               
            </span>
            <span>
              2
            </span>
            <span>
               
              Start
               
            </span>
            <span>
              12:30
            </span>
            <span>
               
              End
               
            </span>
            <span>
              16:00
            </span>
          </div>
        </div>
        <div
          class="c2"
        >
          <header>
            T
          </header>
          <div
            class="card-body"
          >
            <span>
               
              Classes
               
            </span>
            <span>
              0
            </span>
            <span>
               
              Start
               
            </span>
            <span>
              -:-
            </span>
            <span>
               
              End
               
            </span>
            <span>
              -:-
            </span>
          </div>
        </div>
        <div
          class="c2"
        >
          <header>
            F
          </header>
          <div
            class="card-body"
          >
            <span>
               
              Classes
               
            </span>
            <span>
              0
            </span>
            <span>
               
              Start
               
            </span>
            <span>
              -:-
            </span>
            <span>
               
              End
               
            </span>
            <span>
              -:-
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
